test/functional/tests/fault/test_primary_device_error.py

```diff
@@ -0,0 +1,162 @@
+        fio_pid = good_core_fio.run_in_background()
+
+    fio_errors = 0
+    with TestRun.step(
+        "Run fio on error core and check if error count matches with cache stats"
+    ):
+        fio_errors = error_core_fio.run()[0].total_errors()
+
+        if fio_errors == 0:
+            TestRun.fail("No I/O ended with errror!")
+
+    with TestRun.step("Check error statistics on error core"):
+        stats = cache.get_statistics()
+
+        core_errors_in_cache = stats.error_stats.core.total
+        if fio_errors != core_errors_in_cache:
+            TestRun.fail(
+                f"Core errors in cache stats({core_errors_in_cache}) should equal to number of"
+                " fio errors ({fio_errors})"
+            )
+
+    with TestRun.step("Wait for fio on good core"):
+        TestRun.executor.wait_cmd_finish(fio_pid)
+
+    with TestRun.step("Check error statistics on good core"):
+        stats = good_core.get_statistics()
+
+        if stats.error_stats.core.total != 0:
+            TestRun.fail(
+                f"No errors should be reported for good core. "
+                "Actual result: {stats.error_stats.total}"
+            )
+
+    with TestRun.step("Stop the cache"):
+        cache.stop()
+
+    if io_dir == ReadWrite.randwrite:
+        with TestRun.step("Verify error core device contents"):
+            error_core_fio.target(error_core_device).verify_only().run()
+
+    with TestRun.step("Verify good core device contents"):
+        good_core_fio.target(good_core_device).verify_only().run()
+
+
+def prepare_configuration(cache_mode, cache_line_size):
+    cache_device = TestRun.disks["cache"]
+    core_device = TestRun.disks["core"]
+
+    with TestRun.step("Preparing partitions for cache and cores"):
+        cache_device.create_partitions([Size(70, Unit.MebiByte)])
+        core_device.create_partitions(
+            [Size(70, Unit.MebiByte), Size(70, Unit.MebiByte)]
+        )
+        core1 = core_device.partitions[0]
+        core2 = core_device.partitions[1]
+
+    with TestRun.step("Creating error device for one of the cores"):
+        error_device = ErrorDevice(
+            "error",
+            core1,
+            DmTable.uniform_error_table(
+                start_lba=0,
+                stop_lba=int(core1.size.get_value(Unit.Blocks512)),
+                num_error_zones=100,
+                error_zone_size=Size(5, Unit.Blocks512),
+            ).fill_gaps(core1),
+        )
+
+    with TestRun.step("Starting and configuring cache"):
+        cache = casadm.start_cache(
+            cache_device.partitions[0],
+            cache_mode=cache_mode,
+            cache_line_size=cache_line_size,
+            force=True,
+        )
+        result = cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+        if result.exit_code:
+            TestRun.LOGGER.exception("Couldn't set seq cutoff policy")
+        result = cache.set_cleaning_policy(CleaningPolicy.nop)
+        if result.exit_code:
```

Comment:
There's no need for checking err code, if it is not 0 exception will be raised.

Reply:
Fixed